catchpoint-mixin/panels.libsonnet

```diff
@@ -0,0 +1,418 @@
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true')
+        + g.panel.timeSeries.standardOptions.withMax(1)
+        + g.panel.timeSeries.standardOptions.withMin(0)
+        + g.panel.timeSeries.standardOptions.withUnit('percentunit'),
+
+      topAvgConnectionSetupTimeTestName:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Top average connection setup time by tests',
+          targets=[t.topAvgConnectionSetupTimeTestName],
+          description='The top average connection setup time among all test names over the specified interval.'
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true')
+        + g.panel.timeSeries.standardOptions.withUnit('ms'),
+
+      topAvgConnectionSetupTimeNodeName:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Top average connection setup time by nodes',
+          targets=[t.topAvgConnectionSetupTimeNodeName],
+          description='The top average connection setup time among all node names over the specified interval.'
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true')
+        + g.panel.timeSeries.standardOptions.withUnit('ms'),
+
+      topAvgContentLoadingTimeTestName:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Top average content loading time',
+          targets=[t.topAvgContentLoadingTimeTestName],
+          description='The top average content loading time among all test names over the specified interval.'
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true')
+        + g.panel.timeSeries.standardOptions.withUnit('ms'),
+
+      topAvgContentLoadingTimeNodeName:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Top average content loading time by nodes',
+          targets=[t.topAvgContentLoadingTimeNodeName],
+          description='The top average content loading time among all node names over the specified interval.'
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true')
+        + g.panel.timeSeries.standardOptions.withUnit('ms'),
+
+      topAvgRedirectsTestName:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Top average redirects by tests',
+          targets=[t.topAvgRedirectsTestName],
+          description='The top average number of redirects among all test names over the specified interval.'
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true'),
+
+      topAvgRedirectsNodeName:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Top average redirects by nodes',
+          targets=[t.topAvgRedirectsNodeName],
+          description='The top average number of redirects among all node names over the specified interval.'
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true'),
```

Comment:
There's an issue with the legend leaving trailing dashes in this panel:
![image](https://github.com/grafana/jsonnet-libs/assets/1282515/41a11c71-5b76-42ad-8ad9-cdf29d7d75c2)